x/gamm/keeper/migrate_test.go

```diff
@@ -509,18 +575,73 @@ func (suite *KeeperTestSuite) TestUpdateMigrationRecords() {
 			isPreexistingRecordsSet: true,
 			expectErr:               true,
 		},
+		{
+			name: "Mismatch denom0 between the two pools",
+			testingMigrationRecords: []types.BalancerToConcentratedPoolLink{
+				{
+					BalancerPoolId: 1,
+					ClPoolId:       6,
+				},
+			},
+			overwriteBalancerDenom0: "osmo",
+			isPreexistingRecordsSet: false,
+			expectErr:               true,
+		},
+		{
+			name: "Mismatch denom1 between the two pools",
+			testingMigrationRecords: []types.BalancerToConcentratedPoolLink{
+				{
+					BalancerPoolId: 1,
+					ClPoolId:       6,
+				},
+			},
+			overwriteBalancerDenom1: "osmo",
+			isPreexistingRecordsSet: false,
+			expectErr:               true,
+		},
+		{
+			name: "Balancer pool has more than two tokens",
+			testingMigrationRecords: []types.BalancerToConcentratedPoolLink{
+				{
+					BalancerPoolId: 9,
+					ClPoolId:       6,
+				},
+			},
+			isPreexistingRecordsSet:     false,
+			createFourAssetBalancerPool: true,
+			expectErr:                   true,
+		},
 	}
 
 	for _, test := range tests {
 		suite.Run(test.name, func() {
 			suite.SetupTest()
 			keeper := suite.App.GAMMKeeper
 
+			balancerDenom0 := ETH
+			balancerDenom1 := USDC
+
+			if test.overwriteBalancerDenom0 != "" {
+				balancerDenom0 = test.overwriteBalancerDenom0
+			}
+			if test.overwriteBalancerDenom1 != "" {
+				balancerDenom1 = test.overwriteBalancerDenom1
+			}
+
 			// Our testing environment is as follows:
 			// Balancer pool IDs: 1, 2, 3, 4
 			// Concentrated pool IDs: 5, 6, 7, 8
-			suite.PrepareMultipleBalancerPools(4)
-			suite.PrepareMultipleConcentratedPools(4)
+			for i := 0; i < 4; i++ {
+				poolCoins := sdk.NewCoins(sdk.NewCoin(balancerDenom0, sdk.NewInt(1000000000)), sdk.NewCoin(balancerDenom1, sdk.NewInt(5000000000)))
```

Comment:
nit: consider having these coins as varaibles (and then maybe use them across the test cases above?)

Reply:
https://github.com/osmosis-labs/osmosis/pull/4126/commits/447fc5de154cc1913e7e0ebd4742119f6161cbe3